src/cmdlib.sh

```diff
@@ -288,6 +287,45 @@ commit_overlay() {
         --skip-list <(echo /statoverride)
 }
 
+create_content_manifest(){
+    mkdir -p "${workdir}"/tmp/buildinfo
+    base_repos=$(jq .repos "${manifest_tmp_json}")
+    # Make an empty json file in tmp/buildinfo/
+    echo {} > "$workdir"/tmp/buildinfo/content_manifest.json
+
+    # Get the data form content_sets.yaml and map the repo names given in '$base_repos' to their corresponding
+    # pulp repository IDs provided in https://www.redhat.com/security/data/metrics/repository-to-cpe.json
+    python3 -c "
+import json, yaml;
+# Open the yaml and load the data
+f = open('$1')
+data = yaml.safe_load(f);
+f.close();
+repos=[];
+
+for baseRepo in $base_repos:
+    if (baseRepo in data['repo_mapping'].keys()):
+        for i in data['repo_mapping'][baseRepo]:
+            if(i=='name'):
+                data['repo_mapping'][baseRepo][i]=data['repo_mapping'][baseRepo][i].replace('\$ARCH', '$(arch)');
+                repos.append(data['repo_mapping'][baseRepo][i])
+    else:
+        # Printing this warning message till we figure out how to deal with these cases
+        print('Warning: No corresponding repo in repository-to-cpe.json for ' + baseRepo)
```

Comment:
I wonder if we should tweak the semantics a bit: we include those repos that aren't in `repository-to-cpe.json` in `content_sets.yaml` but leave their `name` as `""`. Then here, we can warn and skip if the name is empty, otherwise explicitly error out. This ensures that if the set of manifest repo names changes, we're not just going to forget to update `content_sets.yaml`.

Another way to say this: we require that *every* repo in the manifest must have an entry in `content_sets.yaml`.

Reply:
Agree with it. 
I was thinking we could have two different warning messages; 
- one for the repos with `""` as their `name` in yaml file
- and another one if their is no entry for a `base_repo` in the yaml file itself. This would help us to add an entry to the YAML file in case we miss one in the yaml file in the future.